mobile-widgets/qmlmanager.cpp

```diff
@@ -242,6 +241,8 @@ QMLManager::QMLManager() :
 	// on iOS we should save the data to the DocumentsLocation so it becomes accessible to the user
 	QStringList fileLocations =
 		QStandardPaths::standardLocations(QStandardPaths::DocumentsLocation);
+	// create the object that we use to send emails
+	iosshare = new IosShare(this);
```

Comment:
What's the point of allocating this? Why not make it a simple sub-object of QMLManager? Can it only be allocated later?

If you really want to allocate it, make it a `unique_ptr<IosShare> iosshare` and allocate with `iosshare = std::make_unique<IosShare>();`. Thus, it will be automatically deleted when it goes out of scope (e.g. in the destructor).

Reply:
First I allocated it the support email creation function (and that's where it couldn't be a simple sub-object because it would be deleted before we were done with it) - but that caused us to have multiple instances of it if you tried to send email more than once and that caused Bad Things to happen. So I moved this in the constructor. But I guess here it makes more sense to just use a sub-object...
Simpler, and less code.
Good Catch Number Three